react/src/context/PostHogProvider.tsx

```diff
@@ -45,7 +45,7 @@ export function PostHogProvider({ children, client, apiKey, options }: WithOptio
     // This is used to prevent double initialization when running under React.StrictMode
     // We're not storing a simple boolean here because we want to be able to detect if the
     // apiKey or options have changed.
-    const [alreadyInitialized, setAlreadyInitialized] = useState<AlreadyInitialized>(false)
+    const alreadyInitializedRef = useRef<AlreadyInitialized>(false)
```

Comment:
we say it's not a simple bool then init it to false... should it be `AlreadyInitialized | null` instead?

Reply:
Yeah happy to make that change! (I didn't change this part, but yeah I agree that it'd be neater)

Reply:
the fun of properly reading things for the first time 🙈